src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -4007,6 +4027,17 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
     </Copy>
 
+    <!-- Copy the build product (.dll or .exe). -->
+    <CopyRefAssembly
+        SourcePath="@(IntermediateRefAssembly)"
+        DestinationPath="$(TargetRefPath)"
+        Condition="'$(CopyBuildOutputToOutputDirectory)' == 'true' and '$(SkipCopyBuildProduct)' != 'true' and Exists('@(IntermediateRefAssembly)')"
+            >
+
+      <Output TaskParameter="DestinationPath" ItemName="ReferenceAssembly"/>
```

Comment:
Confirmed that after my last changes `Clean` now deletes the ref assembly from both bin and obj, and `IncrementalClean` leaves it alone. Is there another scenario to test?